### PR TITLE
Event object update

### PR DESCRIPTION
**Pull request by Caleb-Cohen**
# Description

This PR updates the event model to decouple recurring events into separate array objects. It also addresses the storing of an ObjectId for the _id variable of a user instead of the discordId string.

Also added the ability to delete one or delete all events within a given id if the user is the author of the event. 

**This will break any event or user models in your development database**

Great work @romanstetsyk 

## Type of change

Please select everything applicable. Please, do not delete any lines.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Issue
- [ ] Is this related to a specific issue? Is so please specify: #215 

# Checklist:

- [x] I have executed `npm run lint` and resolved any outstanding errors. Most issues can be solved by executing `npm run format`
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: client/src/features/modal/EventModal.jsx [hunk 1]
```diff
@@ -1,17 +1,25 @@
 import React from "react";
 import { GrCalendar } from "react-icons/gr";
 import { FaRegClock } from "react-icons/fa";
-import { IoIosRepeat } from "react-icons/io";
+// import { IoIosRepeat } from "react-icons/io";
```
Comment (Caleb-Cohen) [inline client/src/features/modal/EventModal.jsx:4]: @romanstetsyk, can this be removed?

File: client/src/features/modal/EventModal.jsx [hunk 2]
```diff
@@ -41,18 +65,18 @@ const EventModal = ({ handleClose }) => {
           <p>Description: {modal.activeModal.description}</p>
         </div>
         <div>
-          <section className="flex m-3 gap-1 font-semibold">
+          {/* <section className="flex m-3 gap-1 font-semibold">
```
Comment (Caleb-Cohen) [inline client/src/features/modal/EventModal.jsx:68]: This as well?

**Review by Caleb-Cohen — COMMENTED** [review]
Just a couple comments.

File: server/controllers/events.js [hunk 3]
```diff
@@ -8,28 +8,19 @@ module.exports = {
...
-        discordName: req.user.displayName,
-      });
+      await Event.insertMany(data);
       res.json({ message: "Event created!" });
     } catch (err) {
       console.log(err);
     }
   },
   getAll: async (req, res) => {
+    console.log(req.user);
```
Comment (timmyichen) [inline server/controllers/events.js:20]: extraneous console logs

File: server/controllers/events.js [hunk 4]
```diff
@@ -8,28 +8,19 @@ module.exports = {
...
     try {
-      const events = await Event.find();
+      const events = await Event.find().populate('user').exec();
+      console.log(await events[0]);
```
Comment (timmyichen) [inline server/controllers/events.js:23]: same

File: server/controllers/events.js [hunk 5]
```diff
@@ -8,28 +8,19 @@ module.exports = {
   create: async (req, res) => {
     try {
       let data = JSON.parse(req.body.data);
+      data.forEach(obj => obj.user = req.user._id);
       console.log(data);
```
Comment (timmyichen) [inline server/controllers/events.js:12]: not a part of this PR, but here as well

**Pull request merged**